src/main/groovy/edu/wpi/first/gradlerio/wpi/dependencies/tools/ToolInstallTask.groovy

```diff
@@ -18,36 +18,23 @@ import javax.inject.Inject
 @CompileStatic
 class ToolInstallTask extends DefaultTask {

```

Comment:
Do you have any goal of making this task support being marked as "up-to-date" when nothing has changed?
If so, I'll review this more thoroughly.
Same story if you want to make this task work with the gradle build cache.

If you just want to run this task every time it's invoked, then I won't review as thoroughly.

Reply:
No, we want this task to always get ran.

Reply:
Why? Also, if that's your goal, you should probably document it as such.

Also, can you explain why as well? Either her or in the comment.

I'd assume that once this task has been run, you'd never want to run it unless the inputs or outputs to the task are changed accidentally by someone.

Reply:
Because not everything needs to be incremental. When the tool task gets ran, we want to always ensure the user is getting the latest tool, and don't want gradle to mess us up.

Also, this PR is not the right place to make a change like this anyway. If we were to change this, it would be a separate API.

Reply:
You're trust that gradle won't "mess us up" can be guaranteed by two things, writing tests, and correctly declaring inputs and outputs correctly..

Writing gradle integration tests aren't that difficult. You can just run the same task twice and ensure that the second time it's "up-to-date" but if you change anything, it isn't.

Reply:
Ok. Still wouldn't be in this PR, that would be a separate PR.

Reply:
But I'm also of the opinion that rather then try to get everything perfect, just make gradle always rerun the task. Why is that a bad thing? This is a very not often used task anyway, not used during the regular build.

Reply:
Seems like if you're trying to provide a platform where students should be learning about software in general, it would be good to do things the right way.

Reply:
The right way for this task is to have it always run.

Reply:
Also, is there a way to have an `Input` only be a part of a file, and not the whole thing?

Reply:
Would be good to document as much on the task then probably.

Reply:
Would accept that in a separate PR. Not this one.